BMM.Core/ViewModels/ExploreNewestViewModel.cs

```diff
@@ -157,8 +163,35 @@ private IEnumerable<Document> TranslateDocs(IList<Document> documents)
 
                 sectionHeader.Title = GetText(sectionHeader.TranslationParent, sectionHeader.TranslationId);
             }
+        }
 
-            return documents;
+        private void PrepareCoversCarouselItems(IList<Document> filteredDocs)
+        {
+            var carouselHeaders = filteredDocs
+                .OfType<DiscoverSectionHeader>()
+                .Where(d => d.UseCoverCarousel)
+                .ToList();
+
+            foreach (var carouselHeader in carouselHeaders)
+            {
+                var coverDocuments = new List<CoverDocument>();
+                int currentIndex = filteredDocs.IndexOf(carouselHeader) + 1;
+
+                if (currentIndex >= filteredDocs.Count)
+                    break;
+
+                while (true)
```

Comment:
if the cover carousel is the last section it will throw an index not found exception